frontend/src/component/impact-metrics/ChartConfigModal.tsx

```diff
@@ -0,0 +1,334 @@
+import type { FC } from 'react';
+import { useState, useEffect, useMemo } from 'react';
+import {
+    Dialog,
+    DialogTitle,
+    DialogContent,
+    DialogActions,
+    Button,
+    TextField,
+    Box,
+    Typography,
+    Alert,
+} from '@mui/material';
+import { ImpactMetricsControls } from './ImpactMetricsControls.tsx';
+import {
+    LineChart,
+    NotEnoughData,
+} from '../insights/components/LineChart/LineChart.tsx';
+import { StyledChartContainer } from 'component/insights/InsightsCharts.styles';
+import { useImpactMetricsData } from 'hooks/api/getters/useImpactMetricsData/useImpactMetricsData';
+import { usePlaceholderData } from '../insights/hooks/usePlaceholderData.js';
+import { getDisplayFormat, getTimeUnit, formatLargeNumbers } from './utils.ts';
+import { fromUnixTime } from 'date-fns';
+import { useChartData } from './hooks/useChartData.ts';
+import type { ChartConfig } from './types.ts';
+import type { ImpactMetricsSeries } from 'hooks/api/getters/useImpactMetricsMetadata/useImpactMetricsMetadata';
+
+export interface ChartConfigModalProps {
+    open: boolean;
+    onClose: () => void;
+    onSave: (config: Omit<ChartConfig, 'id'>) => void;
+    initialConfig?: ChartConfig;
+    metricSeries: (ImpactMetricsSeries & { name: string })[];
+    loading?: boolean;
+}
+
+export const ChartConfigModal: FC<ChartConfigModalProps> = ({
+    open,
+    onClose,
+    onSave,
+    initialConfig,
+    metricSeries,
+    loading = false,
+}) => {
+    const [title, setTitle] = useState(initialConfig?.title || '');
```

Comment:
out of the 10+ hooks and methods can they be grouped somehow into higher level hooks?